hawc/apps/animalv2/models.py

```diff
@@ -0,0 +1,576 @@
+                in the comment field below.
+                """,
+    )
+    dtxsid = models.ForeignKey(
+        DSSTox,
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+        verbose_name="DSSTox substance identifier (DTXSID)",
+        related_name="v2_chemicals",
+        help_text=DSSTox.help_text(),
+    )
+    source = models.CharField(max_length=128, verbose_name="Source of chemical", blank=True)
+    purity = models.CharField(max_length=128, verbose_name="Chemical purity", blank=True)
+    vehicle = models.CharField(
+        max_length=64,
+        verbose_name="Chemical vehicle",
+        help_text="Describe vehicle (use name as described in methods but also add the "
+        + "common name if the vehicle was described in a non-standard way). "
+        + 'Enter "not reported" if the vehicle is not described. For inhalation '
+        + "studies, air can be inferred if not explicitly reported. "
+        + 'Examples: "corn oil," "filtered air," "not reported, but assumed clean air."',
+        blank=True,
+    )
+    comments = models.TextField(
+        blank=True,
+        help_text="Additional comments (eg., description, animal husbandry, etc.)",
+    )
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.name
+
+    def get_assessment(self):
+        return self.experiment.get_assessment()
+
+    def get_study(self):
+        return self.experiment.get_study()
+
+    # TODO - flat_complete_header_row / flat_complete_data_row
+
+
+class AnimalGroup(models.Model):
+    objects = managers.AnimalGroupManager()
+
+    # TODO
+    TEXT_CLEANUP_FIELDS = ()
+
+    experiment = models.ForeignKey(
+        Experiment, on_delete=models.CASCADE, related_name="v2_animal_groups"
+    )
+    name = models.CharField(
+        max_length=80,
+        verbose_name="Animal group name",
+        help_text="""
+            Name should be: sex, common strain name, species (plural) and use Title Style
+            (e.g. Male Sprague Dawley Rat, Female C57BL/6 Mice, Male and Female
+            C57BL/6 Mice). For developmental studies, include the generation before
+            sex in title (e.g., F1 Male Sprague Dawley Rat or P0 Female C57 Mice)
+            """,
+    )
+    species = models.ForeignKey(
+        "assessment.Species", related_name="v2_animal_groups", on_delete=models.CASCADE
+    )
+    strain = models.ForeignKey(
+        "assessment.Strain",
+        on_delete=models.CASCADE,
+        related_name="v2_animal_groups",
+        help_text="When adding a new strain, put the stock in parenthesis, e.g., "
+        + '"Sprague-Dawley (Harlan)."',
+    )
+    sex = models.CharField(max_length=1, choices=constants.Sex.choices)
+    animal_source = models.CharField(
+        max_length=128, help_text="Source from where animals were acquired", blank=True
+    )
+    lifestage_at_exposure = models.CharField(
```

Comment:
we have a `animal.constants.Lifestage` enum for this and lifestage_at_assessment - is it ok to use the enum, or was there a reason to use free text?

Reply:
ok to use the enum; I've switched both fields to use that.